test/e2e/run.sh

```diff
@@ -242,135 +243,101 @@ instantiate() { # script API
-	    local POD="$(namespace=kube-system wait_t=${wait_t:-120} vm-wait-pod-regexp nri-resource-policy-)"
-	    if [ ! -z "$POD" ]; then
-		# If the POD contains \n, then the old pod is still there. Wait a sec in this
-		# case and retry.
-		local POD_CHECK=$(echo "$POD" | awk 'BEGIN { RS=""; FS="\n"} { print $2 }')
-		if [ ! -z "$POD_CHECK" ]; then
-		    sleep 1
-		    POD="$(namespace=kube-system wait_t=${wait_t:-60} vm-wait-pod-regexp nri-resource-policy-)"
-		    if [ -z "$POD" ]; then
-			error "Cannot figure out pod name"
-		    fi
-		fi
-
-		if [ "$ds_wait_t" != "none" ]; then
-		    # Wait a while so that the status check can get somewhat meaningful status
-		    vm-command "kubectl -n kube-system rollout status daemonset/nri-resource-policy --timeout=${ds_wait_t:-20s}"
-		    if [ $? -ne 0 ]; then
-			error "Timeout while waiting daemonset/nri-resource-policy to be ready"
-		    fi
-		fi
-
-		# Check if we have anything else than Running status for the pod
-		status="$(vm-command-q "kubectl get pod "$POD" -n kube-system | tail -1 | awk '{ print \$3 }'")"
-		if [ "$status" != "Running" ]; then
-		    # Check if nri-resource-policy failed
-		    if vm-command "kubectl logs $POD -n kube-system | tail -1 | grep -q ^F" 2>&1; then
-			error "Cannot start nri-resource-policy"
-		    fi
-		fi
-
-		vm-command "fuser --kill nri-resource-policy.output.txt 2>/dev/null"
-		vm-command "kubectl -n kube-system logs "$POD" -f >nri-resource-policy.output.txt 2>&1 &"
-
-		vm-port-forward-enable
-	    else
-		error "nri-resource-policy pod not found"
-	    fi
-	    ;;
-
-        *)
-            error "launch: invalid target \"$1\""
+    host-command "$SCP \"$nri_resource_policy_cfg\" $VM_HOSTNAME:" ||
+        command-error "copying \"$nri_resource_policy_cfg\" to VM failed"
+
+    vm-command "helm install --atomic -n kube-system $helm_name ./helm/$plugin \
+             --values=`basename ${nri_resource_policy_cfg}` \
+             --set image.name=localhost/$plugin \
+             --set image.tag=testing \
+             --set image.pullPolicy=Never \
+             --set resources.cpu=50m \
+             --set resources.memory=256Mi \
+             --set plugin.enableTestAPIs=true" ||
+        error "failed to helm install/start plugin $plugin"
+
+    case "$timeout" in
+        ""|"0"|"none")
+            timeout="0"
             ;;
     esac
-    return 0
+
+    if [ -z "$ds_name" ]; then
+        case "$plugin" in
+            *topology*aware*)
+                ds_name=nri-resource-policy-topology-aware
+                ;;
+            *balloons*)
+                ds_name=nri-resource-policy-balloons
+                ;;
+            *)
+                error "Can't wait for plugin $plugin to start, daemonset_name not set"
+                return 0
+                ;;
+        esac
+    fi
+
+    vm-command "kubectl wait -n kube-system ds/${ds_name} --timeout=$timeout \
+             --for=jsonpath='{.status.numberAvailable}'=1" || \
```

Comment:
This turned out to be a source of some flakiness. It seems that DaemonSet may have numberAvailable=1 for a little while when a nri-resource-policy plugin is launched with an illegal configuration. The policy will fail to start with an error, like in balloons test07-maxballoons with the impossible configuration, but helm-launch may report success.

But I don't know if we should grep `up and running` from resource policy logs or what... Any ideas?

Reply:
Once possibility would be to extract the configuration generation once you'v updated it (`kubectl get topologyawareconfigs/default -o jsonpath="{.metadata.generation}"`), then `kubectl wait` for the nodes status' generation in the same object to become the same (and check if the status in there is `"Success"` or `"Failure"`

So maybe something like this...
```bash
$ G=$(kubectl get -n kube-system topologyawareconfigs/default -o jsonpath="{.metadata.generation}")
$ kubectl wait -n kube-system topologyawareconfigs/default --for=jsonpath="{.status.nodes['$(hostname)'].generation}"=$G
$ S=$(kubectl get -n kube-system topologyawareconfigs/default -o jsonpath="{.status.nodes['$(hostname)'].status}")
$ [ "$S" = "Success" ] || {
    E=$(kubectl get -n kube-system topologyawareconfigs/default -o jsonpath="{.status.nodes['$(hostname)'].errors}")
    error "configuration update failed with error \"$E\""
}
```

It is also possible that more seasoned/tougher jsonpath-ninjas are able to reduce this to a fewer set of more incomprehensible commands.

Reply:
@askervin One possibility would be to extract the configuration generation once you'v updated it (`kubectl get topologyawareconfigs/default -o jsonpath="{.metadata.generation}"`), then `kubectl wait` for the nodes status' generation in the same object to become the same (and check if the status in there is `"Success"` or `"Failure"`

So maybe something like this...
```bash
$ G=$(kubectl get -n kube-system topologyawareconfigs/default -o jsonpath="{.metadata.generation}")
$ kubectl wait -n kube-system topologyawareconfigs/default --for=jsonpath="{.status.nodes['$(hostname)'].generation}"=$G
$ S=$(kubectl get -n kube-system topologyawareconfigs/default -o jsonpath="{.status.nodes['$(hostname)'].status}")
$ [ "$S" = "Success" ] || {
    E=$(kubectl get -n kube-system topologyawareconfigs/default -o jsonpath="{.status.nodes['$(hostname)'].errors}")
    error "configuration update failed with error \"$E\""
}
```

It is also possible that more seasoned/tougher jsonpath-ninjas are able to reduce this to a fewer set of more incomprehensible commands.

Reply:
Pushed a slightly cleaner version of the above mentioned idea.